### ISSUE TITLE
False positives for `no-immutable-reactive-statements` when `globalReturn` is enabled

### ISSUE THREAD

**Issue by RedHatter** [sveltejs/eslint-plugin-svelte#1233]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.

### What version of ESLint are you using?

9.27.0

### What version of `eslint-plugin-svelte` are you using?

3.8.1

### What did you do?

<details>
<summary>Configuration</summary>

```
import js from "@eslint/js";
import sveltePlugin from "eslint-plugin-svelte";
import svelteParser from "svelte-eslint-parser";
import tseslint from "typescript-eslint";
export default tseslint.config(
  js.configs.recommended,
  ...tseslint.configs.recommended,
  ...sveltePlugin.configs["flat/recommended"],
  {
    files: ["**/*.svelte", "*.svelte"],
    languageOptions: {
      parser: svelteParser,
      parserOptions: {
        ecmaFeatures: {
          globalReturn: true,
        },
        parser: {
          ts: "@typescript-eslint/parser",
        },
        extraFileExtensions: [".svelte"],
      },
    }
  },
  {
    files: ["**/*.svelte.ts", "*.svelte.ts"],
    languageOptions: {
      parser: svelteParser,
      parserOptions: {
        parser: tseslint.parser,
      },
    }
  },
  {
    files: ["**/*.svelte.js", "*.svelte.js"],
    languageOptions: {
      parser: svelteParser,
    }
  },
);
```
</details>

```svelte
<script lang="ts">
  export let a;

  $: b = a;
</script>

{b}
```

### What did you expect to happen?

I expected the code to pass eslint

### What actually happened?

Eslint fails with `svelte/no-immutable-reactive-statements` error

### Link to **GitHub Repo** with Minimal Reproducible Example

https://eslint-online-playground.netlify.app/#eNq1VMFu2zAM/RVC2GELEvvutsOAbb0NG3addlBs2nAmS4YkFymC/PsoS3LsJOuKFr0YFvneI0WRPDBryvzb42fd9Vqhcpl9QOmQFezWlqbtHUihmjvOnOXsI1cAuO+1ITM6EDdcedO7ArZwNx5v80AjKFeH7ZErtmZoZUvSpVZ122Q7S+otxSOVnYXa6A44+xRA+Y7ikE70h2x+yKFpVUIG4KYfjZuAuMIRxqJJnGDbJOrom3OcDa6Ed489hotEzgiOV6+wFoM8keLF3vtS7Gw82cxgqbsOVYXV2ruyLDtjXMXMr5xwvzirpXD5DM7Z75Fx8B+AupVoCyDgapWvoghna7rL6RQYMD7pIBr83rtWK6JFEYBQmWJRw0hKzksS9UTZiXsUbjA+iZkDoJF6K+RPJJ8qwJkBJz2A4+w/hV6wHclRb1w8R56e8B9iuHdG3FNNvu4dKhsSpuJclOJEo1ZNh6dqSm+4LOtoeLPKJtrUOcHw4uz9eC2y94ZXZT9F/nBDo96L8g8pkKxWNOajQBgnzvxTdroapG/M4Kjw4Qv2vqFV2SKlMsVcbgRPpQFAS4MY46ZF8KTzfEtchT7Deb45rkJPXfpfwLRV5rixkFwd2fEvFSTV7A==

### Additional comments

When the eslint option `languageOptions.parserOptions.ecmaFeatures.globalReturn` is set `no-immutable-reactive-statements` fails for most reactive statements. Not all of them but I haven't been able to determine why some work and others don't.
